<a href="https://colab.research.google.com/github/ornellajogi/courses/blob/main/Kaggle_Intro_to_programming.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Source: https://www.kaggle.com/code/alexisbcook/functions

Question 1
Next, you will run some code from the tutorial, so you can see how it works for yourself. Run the next code cell without changes.

In [2]:
print("Hello, world!")

Hello, world!


You just ran code to print Hello world!, which you should see in the output above.

The second line of code (q1.check()) checks your answer. You should never modify this checking code; if you remove it, you won't get credit for completing the problem.

Question 2
Now, you will print another message of your choosing. To do this, change print("Your message here!") to use a different message. For instance, you might like to change it to something like:

print("Good morning!")
print("I am learning how to code :D")
Or, you might like to see what happens if you write something like print("3+4"). Does it return 7, or does it just think of "3+4" as just another message?

Make sure that your message is enclosed in quotation marks ("), and the message itself does not use quotation marks. For instance, this will throw an error: print("She said "great job" and gave me a high-five!") because the message contains quotation marks. If you decide to take the Python course after completing this course, you will learn more about how to avoid this error in Lesson 6.

Feel free to try out multiple messages!

In [3]:
# TODO: Change the message
print("Data Science Star in the making")

Data Science Star in the making


Question 3
As you learned in the tutorial, a comment in Python has a pound sign (#) in front of it, which tells Python to ignore the text after it.

Putting a pound sign in front of a line of code will make Python ignore that code. For instance, this line would be ignored by Python, and nothing would appear in the output:

#print(1+2)
Removing the pound sign will make it so that you can run the code again. When we remove the pound sign in front of a line of code, we call this uncommenting.

In this problem, you will uncomment two lines in the code cell below and view the output:

Remove the # in front of q3.hint(). To avoid errors, do NOT remove the # in front of # Uncomment to view hint.
Next, remove the # in front of q3.solution().
As in the previous questions, do not change the final line of code that marks your work as completed.

Question 4
In the tutorial, you defined several variables to calculate the total number of seconds in a year. Run the next code cell to do the calculation here.

In [4]:
# Create variables
num_years = 4
days_per_year = 365
hours_per_day = 24
mins_per_hour = 60
secs_per_min = 60

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126144000


Use the next code cell to:

Define a variable births_per_min and set it to 250. (There are on average 250 babies born each minute.)
Define a variable births_per_day that contains the average number of babies born each day. (To set the value of this variable, you should use births_per_min and some of the variables from the previous code cell.)
Remember you can always get a hint if you need it!

In [6]:
# TODO: Set the value of the births_per_min variable
births_per_min = 250

# TODO: Set the value of the births_per_day variable
births_per_day = births_per_min * (hours_per_day * mins_per_hour)

print(births_per_day)

360000


🌶️ Question 5
(Questions marked with a 🌶️ will be a little bit more challenging than the others! Remember you can always get a hint or view the solution.)

The Titanic competition is Kaggle's most famous data science competition. In this competition, participants are challenged to build a machine learning model that can predict whether or not passengers survived the Titanic shipwreck, based on information like age, sex, family size, and ticket number.

Run the next code cell without changes to load and preview the titanic data.

Don't worry about the details of the code for now - the end result is just that the all of the titanic data has been loaded in a variable named titanic_data. (In order to learn how to write this code yourself, you can take the Python course and then the Pandas course.)

In [9]:
# Load the data from the titanic competition
import pandas as pd
titanic_data = pd.read_csv("train.csv")

#Show the first five rows of the data
titanic_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

The data has a different row for each passenger.

The next code cell defines and prints the values of three variables:

total = total number of passengers who boarded the ship
survived = number of passengers who survived the shipwreck
minors = number of passengers under 18 years of age
Run the code cell without changes. (Don't worry about the details of how these variables are calculated for now. You can learn more about how to calculate these values in the Pandas course.)

In [11]:
# Number of total passengers
total = len(titanic_data)
print(total)

# Nr of passengers who survived
survived = (titanic_data.Survived == 1).sum()
print(survived)

# Nr of passengers under 18
minors = (titanic_data.Age < 18).sum()
print(minors)

891
342
113


So,

total = 891 (there were 891 passengers on board the Titanic),
survived = 342 (342 passengers survived), and
minors = 113 (113 passengers were under the age of 18).
In the code cell below, replace the underlines (____) with code to calculate the values for two more variables:

survived_fraction should be set to the fraction of passengers who survived the Titanic disaster.
minors_fraction should be the fraction of passengers who were minors (under the age of 18).
For each variable, your answer should be a number between 0 and 1.

If you need a hint or want to view the solution, you can skip to the next code cell and uncomment the appropriate lines of code (q5.hint() and q5.solution()).

In [12]:
survived_fraction = survived / total

print(survived_fraction)

minors_fraction = minors / total

print(minors_fraction)

0.3838383838383838
0.12682379349046016


LESSON: FUNCTIONS

Introduction
In this lesson, you will learn how to organize your code with functions. A function is a block of code designed to perform a specific task. As you'll see, functions will let you do roughly the same calculation multiple times without duplicating any code.

Intro to functions: a simple example
We begin with a simple example of a function. The add_three() function below accepts any number, adds three to it, and then returns the result.

In [13]:
# Define the function
def add_three(input_var):
    output_var =input_var + 3
    return output_var

Every function is composed of two pieces: a header and body.



Header
The function header defines the name of the function and its argument(s).

Every function header begins with def, which tells Python that we are about to define a function.
In the example, the function name is add_three.
In the example, the argument is input_var. The argument is the name of the variable that will be used as input to the function. It is always enclosed in parentheses that apppear immediately after the name of the function. (Note that a function can also have no arguments, or it can have multiple arguments. You'll see some examples of this later in the lesson.)
For every function, the parentheses enclosing the function argument(s) must be followed by a colon :.
Body
The function body specifies the work that the function does.

Every line of code in the function body must be indented exactly four spaces. You can do this by pushing the space bar four times, or by hitting the "Tab" button once on your keyboard. (As you learn more about Python, you may need to indent your code by more than four spaces, but you'll learn more about that later in this course.)
The function does its work by running all of the indented lines from top to bottom.
It takes the argument as input, which in the example is input_var.
The function creates a new variable output_var with the calculation output_var = input_var + 3.
Then, the final line of code, called the return statement, just returns the value in output_var as the function's output.
The code cell above just defines the function, but does not run it. The details of the function body will make more sense after the next code cell, when we actually run the function.

How to run (or "call") a function¶
When we run a function, it can also be referred to as "calling" the function.

In the code cell below, we run the function with 10 as the input value. We define a new variable new_number which is set to the output of the function.